js/gestionPresupuesto.js

```diff
@@ -1,25 +1,148 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
 
 // TODO: Variable global
+let presupuesto = 0;
+let gastos = [];
+let idGasto = 0;
 
+function actualizarPresupuesto(importe) {
+    if (importe < 0 || isNaN(importe) == true) {
+        console.log("El valor ingresado no es válido")
+        return -1;
+    }
+    presupuesto = importe;
 
-function actualizarPresupuesto() {
-    // TODO
+    return presupuesto;
 }
 
+
 function mostrarPresupuesto() {
-    // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`;
+}
+
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+    if (valor < 0 || isNaN(valor)) {
+        this.valor = 0;
+    } else {
+        this.valor = valor;
+    }
+    this.descripcion = descripcion;
+
+    //Para la verificación de la fecha lei en MDM web docs que si el valor de Date.parse falla se devuelve NaN
+    if (!fecha || isNaN(Date.parse(fecha))) {
+        this.fecha = Date.now();
+    } else {
+        this.fecha = new Date(fecha).getTime();
+    }
+    //Verifica que la etiqueta no este vacia
+    if (!etiquetas || etiquetas.length === 0) {
+        this.etiquetas = [];
+    } else {
+        this.etiquetas = etiquetas;
```

Comment:
En lugar de asignar directamente el array de etiquetas, también pudes utilizar el método `this.anyadirEtiquetas`. Así evitas duplicados.

Reply:
Gracias por al corrección! Claro de esa forma usamos el mismo método que ya verifica duplicados. Lo refactorizo para la próxima entrega.